### DISCUSSION TITLE
Timeout on Stored Procedure that takes a little over a minute to run

### DISCUSSION THREAD

**Discussion by acarlso1** [dnnsoftware/Dnn.Platform#5726] [Q&A]
I've got a Stored Procedure but it can sometimes take a minute or a little more to complete. I can't seem to find a way in DAL2 to extended the timeout which seems to be locked at 30sec. I've tried setting the SqlCommandTimeOut in the web.config but didn't seem to change anything.

<add key="SqlCommandTimeOut" value="500"/>

using (IDataContext ctx = DataContext.Instance())
{
    ctx.Execute(System.Data.CommandType.StoredProcedure, "dbo.ChangeGroupCode", oldCode, newCode);
}

Thanks, any help is appreciated

**Comment by bdukes** [comment]
There is not currently an exposed way to adjust the timeout. [PetaPoco added a mechanism for setting the timeout in 2011](https://github.com/CollaboratingPlatypus/PetaPoco/issues/37), but DNN's `DataContext` does not currently expose that option. I don't see a way to adjust that other than via reflection.

**Comment by sleupold** [comment]
Did you consider improving indexes or using indexed queries?
